pkg/cloud/libvirt/client/client.go

```diff
@@ -162,6 +162,12 @@ func (client *libvirtClient) Close() error {
 		glog.Infof("Error closing libvirt connection: %v", err)
 	}
 
+	glog.Infof("Freeing the client pool")
```

Comment:
Since the storage pool keeps a reference on the libvirt connection, I would free it before closing the connection in case it needs to use the connection to do some cleanups.

Reply:
Actually, it's not that important because `virConnectClose` is mostly `virConnectUnref`, so the connection won't be really closed until the storage pool is freed too (which is what this bug is about after all :)
One nice thing about freeing the pool first is that we can check `client.connection.Close()` return value to warn about leaks in the future https://libvirt.org/html/libvirt-libvirt-host.html#virConnectClose

> a positive number if at least 1 reference remains on success. The returned value should not be assumed to be the total reference count. A return of 0 implies no references remain and the connection is closed and memory has been freed.

Reply:
This is really good to know!

Reply:
I added this in https://github.com/cfergeau/cluster-api-provider-libvirt/commits/leaks